File: decompiled/3.18.0/be/ehealth/businessconnector/addressbook/service/impl/AddressbookTokenServiceImpl.java
package be.ehealth.businessconnector.addressbook.service.impl;

import be.ehealth.businessconnector.addressbook.service.AddressbookTokenService;
import be.ehealth.businessconnector.addressbook.service.TokenServiceFactory;
import be.ehealth.technicalconnector.exception.TechnicalConnectorException;
import be.ehealth.technicalconnector.exception.TechnicalConnectorExceptionValues;
import be.ehealth.technicalconnector.service.sts.security.SAMLToken;
import be.ehealth.technicalconnector.validator.EhealthReplyValidator;
import be.ehealth.technicalconnector.validator.SessionValidator;
import be.ehealth.technicalconnector.ws.ServiceFactory;
import be.ehealth.technicalconnector.ws.domain.GenericRequest;
import be.fgov.ehealth.addressbook.protocol.v1.GetOrganizationContactInfoRequest;
import be.fgov.ehealth.addressbook.protocol.v1.GetOrganizationContactInfoResponse;
import be.fgov.ehealth.addressbook.protocol.v1.GetProfessionalContactInfoRequest;
import be.fgov.ehealth.addressbook.protocol.v1.GetProfessionalContactInfoResponse;
import be.fgov.ehealth.addressbook.protocol.v1.SearchOrganizationsRequest;
import be.fgov.ehealth.addressbook.protocol.v1.SearchOrganizationsResponse;
import be.fgov.ehealth.addressbook.protocol.v1.SearchProfessionalsRequest;
import be.fgov.ehealth.addressbook.protocol.v1.SearchProfessionalsResponse;
import be.fgov.ehealth.commons.protocol.v2.RequestType;
import be.fgov.ehealth.commons.protocol.v2.StatusResponseType;
import java.util.HashMap;
import java.util.Map;
import javax.xml.soap.SOAPException;

public class AddressbookTokenServiceImpl implements AddressbookTokenService {
   private SessionValidator sessionValidator;
   private EhealthReplyValidator ehealthReplyValidator;
   private static Map<Class<? extends StatusResponseType>, String> soapActions = new HashMap();

   public AddressbookTokenServiceImpl(SessionValidator sessVal, EhealthReplyValidator ehRepVal) throws TechnicalConnectorException {
      this.sessionValidator = sessVal;
      this.ehealthReplyValidator = ehRepVal;
   }

   public GetOrganizationContactInfoResponse getOrganizationContactInfo(SAMLToken token, GetOrganizationContactInfoRequest request) throws TechnicalConnectorException {
      return (GetOrganizationContactInfoResponse)this.invoke(token, request, GetOrganizationContactInfoResponse.class);
   }

   public GetProfessionalContactInfoResponse getProfessionalContactInfo(SAMLToken token, GetProfessionalContactInfoRequest request) throws TechnicalConnectorException {
      return (GetProfessionalContactInfoResponse)this.invoke(token, request, GetProfessionalContactInfoResponse.class);
   }

   public SearchOrganizationsResponse searchOrganizations(SAMLToken token, SearchOrganizationsRequest request) throws TechnicalConnectorException {
      return (SearchOrganizationsResponse)this.invoke(token, request, SearchOrganizationsResponse.class);
   }

   public SearchProfessionalsResponse searchProfessionals(SAMLToken token, SearchProfessionalsRequest request) throws TechnicalConnectorException {
      return (SearchProfessionalsResponse)this.invoke(token, request, SearchProfessionalsResponse.class);
   }

   private <T extends StatusResponseType> T invoke(SAMLToken token, RequestType request, Class<T> clazz) throws TechnicalConnectorException {
      try {
         this.sessionValidator.validateToken(token);
         GenericRequest service = TokenServiceFactory.getService(token);
         service.setPayload((Object)request);
         service.setSoapAction((String)soapActions.get(clazz));
         T response = (StatusResponseType)ServiceFactory.getGenericWsSender().send(service).asObject(clazz);
         this.ehealthReplyValidator.validateReplyStatus(response);
         return response;
      } catch (SOAPException var6) {
         throw new TechnicalConnectorException(TechnicalConnectorExceptionValues.ERROR_WS, var6, new Object[]{var6.getMessage()});
      }
   }

   static {
      soapActions.put(GetProfessionalContactInfoResponse.class, "urn:be:fgov:ehealth:addressbook:protocol:v1:getProfessionalContactInfo");
      soapActions.put(GetOrganizationContactInfoResponse.class, "urn:be:fgov:ehealth:addressbook:protocol:v1:getOrganizationContactInfo");
      soapActions.put(SearchProfessionalsResponse.class, "urn:be:fgov:ehealth:addressbook:protocol:v1:searchProfessionals");
      soapActions.put(SearchOrganizationsResponse.class, "urn:be:fgov:ehealth:addressbook:protocol:v1:searchOrganizations");
   }
}

File: src/main/kotlin/org/taktik/connector/business/addressbook/service/impl/AddressbookTokenServiceImpl.kt
/*
 *
 * Copyright (C) 2018 Taktik SA
 *
 * This file is part of FreeHealthConnector.
 *
 * FreeHealthConnector is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation.
 *
 * FreeHealthConnector is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with FreeHealthConnector.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

package org.taktik.connector.business.addressbook.service.impl

import org.taktik.connector.business.addressbook.service.AddressbookTokenService
import org.taktik.connector.business.addressbook.service.TokenServiceFactory
import org.taktik.connector.technical.exception.TechnicalConnectorException
import org.taktik.connector.technical.exception.TechnicalConnectorExceptionValues
import org.taktik.connector.technical.service.sts.security.SAMLToken
import org.taktik.connector.technical.validator.EhealthReplyValidator
import org.taktik.connector.technical.validator.SessionValidator
import org.taktik.connector.technical.ws.ServiceFactory
import org.taktik.connector.technical.ws.domain.GenericRequest
import be.fgov.ehealth.addressbook.protocol.v1.GetOrganizationContactInfoRequest
import be.fgov.ehealth.addressbook.protocol.v1.GetOrganizationContactInfoResponse
import be.fgov.ehealth.addressbook.protocol.v1.GetProfessionalContactInfoRequest
import be.fgov.ehealth.addressbook.protocol.v1.GetProfessionalContactInfoResponse
import be.fgov.ehealth.addressbook.protocol.v1.SearchOrganizationsRequest
import be.fgov.ehealth.addressbook.protocol.v1.SearchOrganizationsResponse
import be.fgov.ehealth.addressbook.protocol.v1.SearchProfessionalsRequest
import be.fgov.ehealth.addressbook.protocol.v1.SearchProfessionalsResponse
import be.fgov.ehealth.commons.protocol.v2.RequestType
import be.fgov.ehealth.commons.protocol.v2.StatusResponseType
import java.util.HashMap
import javax.xml.soap.SOAPException

class AddressbookTokenServiceImpl @Throws(TechnicalConnectorException::class)
constructor(private val sessionValidator: SessionValidator, private val ehealthReplyValidator: EhealthReplyValidator) : AddressbookTokenService {

    @Throws(TechnicalConnectorException::class)
    override fun getOrganizationContactInfo(token: SAMLToken, request: GetOrganizationContactInfoRequest): GetOrganizationContactInfoResponse {
        return this.invoke(token, request, GetOrganizationContactInfoResponse::class.java)
    }

    @Throws(TechnicalConnectorException::class)
    override fun getProfessionalContactInfo(token: SAMLToken, request: GetProfessionalContactInfoRequest): GetProfessionalContactInfoResponse {
        return this.invoke(token, request, GetProfessionalContactInfoResponse::class.java)
    }

    @Throws(TechnicalConnectorException::class)
    override fun searchOrganizations(token: SAMLToken, request: SearchOrganizationsRequest): SearchOrganizationsResponse {
        return this.invoke(token, request, SearchOrganizationsResponse::class.java)
    }

    @Throws(TechnicalConnectorException::class)
    override fun searchProfessionals(token: SAMLToken, request: SearchProfessionalsRequest): SearchProfessionalsResponse {
        return this.invoke(token, request, SearchProfessionalsResponse::class.java)
    }

    @Throws(TechnicalConnectorException::class)
    private operator fun <T : StatusResponseType> invoke(token: SAMLToken, request: RequestType, clazz: Class<T>): T {
        try {
            this.sessionValidator.validateToken(token)
            val service = TokenServiceFactory.getService(token).apply {
                setPayload(request)
                setSoapAction(soapActions[clazz])
            }
            val response = ServiceFactory.getGenericWsSender().send(service).asObject(clazz)
            this.ehealthReplyValidator.validateReplyStatus(response)
            return response
        } catch (ex: SOAPException) {
            throw TechnicalConnectorException(TechnicalConnectorExceptionValues.ERROR_WS, ex, ex.message)
        }

    }

    companion object {
        private val soapActions = HashMap<Class<out StatusResponseType>, String>()

        init {
            soapActions.put(GetProfessionalContactInfoResponse::class.java, "urn:be:fgov:ehealth:addressbook:protocol:v1:getProfessionalContactInfo")
            soapActions.put(GetOrganizationContactInfoResponse::class.java, "urn:be:fgov:ehealth:addressbook:protocol:v1:getOrganizationContactInfo")
            soapActions.put(SearchProfessionalsResponse::class.java, "urn:be:fgov:ehealth:addressbook:protocol:v1:searchProfessionals")
            soapActions.put(SearchOrganizationsResponse::class.java, "urn:be:fgov:ehealth:addressbook:protocol:v1:searchOrganizations")
        }
    }
}
